src/role_squadheal.js

```diff
@@ -10,13 +10,37 @@ roles.squadheal = {};
 
 roles.squadheal.settings = {
   layoutString: 'MH',
-  amount: [17, 17],
+  amount: [1, 1],
   fillTough: true,
 };
 
+roles.squadheal.healClosestCreep = function(creep) {
+  const myCreep = creep.pos.findClosestByRange(FIND_MY_CREEPS, {
+    filter: function(object) {
+      if (object.hits < object.hitsMax) {
+        return true;
+      }
+      return false;
+    },
+  });
+  if (myCreep !== null) {
+    creep.say('heal', true);
+    const range = creep.pos.getRangeTo(myCreep);
+    if (range <= 1) {
+      creep.heal(myCreep);
+    } else {
+      creep.moveTo(myCreep);
+      creep.rangedHeal(myCreep);
+    }
+    return true;
+  }
+  return false;
+};
```

Comment:
Creep.proto healMyCreeps may be used here. just need adapt it for findClosest with a new arg.

Reply:
Yes, true. Don't want to touch it here.